Automattic/jetpack
# Protect: Introduce navigation related components (`Navigation`, `NavigationItem` and `NavigationGroup`)

Introduce all components related to navigation inside `Protect`. 

It was introduced 3 components: `Navigation`, `NavigationItem`, and `NavigationGroup`.

The component was designed to work with a mouse and keyboard (tab and arrows navigation).

#### Changes proposed in this Pull Request:

* Introduce 3 new components used to compose left `Navigation` inside `Protect`.

#### Jetpack product discussion


p1HpG7-fBP-p2

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Start `Storybook`.
* Navigate to `Protect/Components/Navigation`.
* You should see the Navigation example.
* Navigate with it using: Mouse and Keyboard (Arrows and Tab - Using enter and space as "click").
* Disabled items or Group items should not be clickable/selectable.

#### Demo

https://user-images.githubusercontent.com/1663717/163895028-bf803951-1d61-4760-a185-a90cc77b4366.mp4
